### PR TITLE
feat💥: Improve channel permissions (see desc.)

### PR DESCRIPTION
**Pull request by leestarb**
## What type of pull request is this?

- [ ] Non-breaking code change
- [x] Breaking code change
- [x] Documentation change/addition

## Description

Some improvements for channel permissions related stuff
I'm open to any improvements

## Changes

- Implement `DictSerializationMixin` in `PermissionOverwrite`
- Rewrite `Channel.permission_overwrites` as list
- Make `Channel.permission_overwrites` public
- Documentate `Channel.permission_overwrites`
- Use docs for attr attributes in PermissionOverwrite
- Add a note for `id` attribute in PermissionOverwrite
- Remove useless `_original_permission_overwrites`
- Use `map`, `to_dict` for permissions in create channel methods
- Also use correct check for a empty list
- Add `Channel.permissions_for` helper method
- Add a note about it in `Member.channel_permissions`
- Refactor `Member.channel_permissions` to improve performance

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/models/discord/user.py [hunk 1]
```diff
@@ -386,8 +386,9 @@ def channel_permissions(self, channel: "TYPE_GUILD_CHANNEL") -> Permissions:
         if Permissions.ADMINISTRATOR in permissions:
             return Permissions.ALL
 
+        overwrites = {overwrite.id: overwrite for overwrite in channel.permission_overwrites}
```
Comment (leestarb) [inline dis_snek/models/discord/user.py:389]: Does anyone know how to optimise this?

File: dis_snek/models/discord/channel.py [hunk 2]
```diff
@@ -776,13 +778,53 @@ def category(self) -> Optional["GuildCategory"]:
...
+                        permissions &= ~overwrite.deny
+                        permissions |= overwrite.allow
+                        break
+
+                return permissions
+
+        else:
+            instance = to_snowflake(instance)
+            instance = self._client.cache.get_member(self._guild_id, instance) or self._client.cache.get_role(instance)
+            # Will be replaced with `Guild.get_` just after #289 merge!
```
Comment (leestarb) [inline dis_snek/models/discord/channel.py]: Do not merge this PR unless its done!

Comment (leestarb) [reply dis_snek/models/discord/channel.py]: Done

**Review by LordOfPolls — APPROVED** [review]

**Pull request merged**